cli/unstable_ansi.ts

```diff
@@ -0,0 +1,961 @@
+ * import * as Ansi from "@std/cli/unstable_ansi";
+ *
+ * console.log(Ansi.doubleHeightTop + "Hello World!");
+ * console.log(Ansi.doubleHeightBottom + "Hello World!");
+ * await delay(1000);
+ * console.log(
+ *   Ansi.moveCursorUpStart(2) +
+ *     Ansi.deleteLines(1) +
+ *     Ansi.singleWidth,
+ * );
+ * ```
+ */
+export const singleWidth = "\x1b#5";
+
+/**
+ * Causes content on the current line to stretch out, with each character
+ * taking up two columns. Can be reverted with {@linkcode Ansi.singleWidth}.
+ *
+ * @returns string The ANSI escape code for double-width.
+ *
+ * @example Usage
+ * ```ts ignore
+ * import * as Ansi from "@std/cli/unstable_ansi";
+ *
+ * console.log(Ansi.doubleWidth + "Hello World!");
+ * ```
+ */
+export const doubleWidth = "\x1b#6";
+
+/**
+ * Saves current cursor position that can later be restored with
+ * {@linkcode Ansi.restoreCursorPosition}.
+ *
+ * @returns string The ANSI escape code for saving cursor position.
+ *
+ * @example Usage
+ * ```ts ignore
+ * import * as Ansi from "@std/cli/unstable_ansi";
+ *
+ * console.log(
+ *   Ansi.saveCursorPosition +
+ *     Ansi.setCursorPosition(Deno.consoleSize().rows, 1) +
+ *     Ansi.eraseLine +
+ *     "Hello World!" +
+ *     Ansi.restoreCursorPosition,
+ * );
+ * ```
+ */
+export const saveCursorPosition = "\x1b7";
+
+/**
+ * Restores cursor position that was earlier saved with
+ * {@linkcode Ansi.saveCursorPosition}.
+ *
+ * @returns string The ANSI escape code for restoring cursor position.
+ *
+ * @example Usage
+ * ```ts ignore
+ * import * as Ansi from "@std/cli/unstable_ansi";
+ *
+ * console.log(
+ *   Ansi.saveCursorPosition +
+ *     Ansi.setCursorPosition(Deno.consoleSize().rows) +
+ *     Ansi.eraseLine +
+ *     "Hello World!" +
+ *     Ansi.restoreCursorPosition,
+ * );
+ * ```
+ */
+export const restoreCursorPosition = "\x1b8";
+
+/**
+ * This is a full reset of the terminal, reverting it back to its original
+ * default settings, clearing the screen, resetting modes, colors, character
+ * sets and more. Essentially making the terminal behave as if it were just
+ * started by the user. This command is very disruptive to the user. Also see
+ * {@linkcode Ansi.softReset}.
+ *
+ * @returns string The ANSI escape code for hard resetting the terminal.
+ *
+ * @example Usage
+ * ```ts ignore
+ * import { delay } from "@std/async/delay";
+ * import * as Ansi from "@std/cli/unstable_ansi";
+ *
+ * console.log(Ansi.hideCursor);
+ * await delay(5000);
+ * console.log(Ansi.hardReset);
+ * ```
+ */
+export const hardReset = "\x1bc";
+
+/**
+ * This command resets many settings to their initial state without fully
+ * reinitializing the terminal like {@linkcode Ansi.hardReset}. It preserves
+ * things like cursor position and display content, but clears modes,
+ * character sets, etc. Should probably be called when exiting the program.
+ *
+ * @returns string The ANSI escape code for soft resetting the terminal.
```

Comment:
```suggestion
 * @returns The ANSI escape code for soft resetting the terminal.
```
No need to declare the return type in JSDocs thanks to TypeScript. Ditto elsewhere.